src/main/java/org/openrewrite/java/migrate/apache/commons/lang/ApacheCommonsStringUtils.java

```diff
@@ -0,0 +1,365 @@
+        String after(String s) {
+            return s.substring(0, s.length() - 1);
+        }
+    }
+
+    public static class DefaultString {
+        @BeforeTemplate
+        String before(String s) {
+            return StringUtils.defaultString(s);
+        }
+
+        @AfterTemplate
+        String after(String s) {
+            return Objects.toString(s);
+        }
+    }
+
+    public static class Equals {
+        @BeforeTemplate
+        boolean before(String s, String other) {
+            return StringUtils.equals(s, other);
+        }
+
+        @AfterTemplate
+        boolean after(String s, String other) {
+            return Objects.equals(s, other);
+        }
+    }
+
+    @SuppressWarnings("ConstantValue")
+    public static class IsEmpty {
+        @BeforeTemplate
+        boolean before(String s) {
+            return StringUtils.isEmpty(s);
+        }
+
+        @AfterTemplate
+        boolean after(String s) {
+            return s == null || s.isEmpty();
+        }
+    }
+
+    public static class Replace {
+        @BeforeTemplate
+        String before(String s, String target, String replacement) {
+            return StringUtils.replace(s, target, replacement);
+        }
+
+        @AfterTemplate
+        String after(String s, String target, String replacement) {
+            return s.replaceAll(target, replacement);
+        }
+    }
+
+    public static class Split {
+        @BeforeTemplate
+        String[] before(String s) {
+            return StringUtils.split(s);
+        }
+
+        @AfterTemplate
+        String[] after(String s) {
+            return s.split(" ");
+        }
+    }
+
+    public static class StripEnd {
+        @BeforeTemplate
+        String before(String s, String suffix) {
+            return StringUtils.stripEnd(s, suffix);
+        }
+
+        @AfterTemplate
+        String after(String s, String suffix) {
+            return s.endsWith(suffix) ? s.substring(0, s.lastIndexOf(suffix)) : s;
+        }
+    }
+
+    public static class Strip {
+        @BeforeTemplate
+        String before(String s) {
+            return StringUtils.strip(s);
+        }
+
+        @AfterTemplate
+        String after(String s) {
+            return s.trim();
+        }
+    }
+
+    // TODO: These two methods don't generate the recipe templates right
+    //public static class LeftPad {
+    //    @BeforeTemplate
+    //    String before(String s, int l) {
+    //        return StringUtils.leftPad(s, l);
+    //    }
+
+    //    @AfterTemplate
+    //    String after(String s, int l) {
+    //        return String.format("%" + l + "s", s);
+    //    }
+    //}
+
+    //public static class RightPad {
+    //    @BeforeTemplate
+    //    String before(String s, int l) {
+    //        return StringUtils.rightPad(s, l);
+    //    }
+
+    //    @AfterTemplate
+    //    String after(String s, int l) {
+    //        return String.format("%" + (-l) + "s", s);
+    //    }
+    //}
```

Comment:
I think we can try these again after the recent fixes.

Reply:
Seems like there still is an issue with these recipes. The error I get is `error: unclosed string literal .builder("String.format(\"%\" #{any(int)} l + \"s#{any(java.lang.String)}", s)");`

Reply:
Seems like the last '"' after `String)}` should also be escaped

Reply:
Well it happens every time I regenerate the recipes so I assume its something in rewrite-templating. This is on my desktop which didn't ever have the local rewrite-templating dependency so I don't think it could be an issue with that. If you uncomment the code and try to rebuild do you see the error?

Reply:
Confirmed; we can ignore this one for now then, and focus on the others to get a first batch through.